flink-runtime/src/main/java/org/apache/flink/runtime/util/ZooKeeperUtils.java

```diff
@@ -402,6 +440,87 @@ public static ZooKeeperLeaderElectionDriverFactory createLeaderElectionDriverFac
         return new ZooKeeperLeaderElectionDriverFactory(client, path);
     }
 
+    public static void writeLeaderInformationToZooKeeper(
+            LeaderInformation leaderInformation,
+            CuratorFramework curatorFramework,
+            BooleanSupplier hasLeadershipCheck,
+            String connectionInformationPath)
+            throws Exception {
+        final byte[] data;
+
+        if (leaderInformation.isEmpty()) {
+            data = null;
```

Comment:
can we add a test case to `ZooKeeperUtilsTest` for writing data (including `null`) into the path? I did some digging into the Curator4 code tot check: It looks like it is. But it's hard to find information here. The [JavaDoc](https://www.javadoc.io/doc/org.apache.curator/curator-framework/4.0.0/org/apache/curator/framework/api/PathAndBytesable.html) is also not helpful. WDYT? Is it worth the effort?

Reply:
Do you mean a test where we write and read a `LeaderInformation.empty()`? This sounds like a good idea. If you meant a test where we use curator to write a `null` data, I think this not needed.

Reply:
Yup, agreed 👍